### ISSUE TITLE
Comparing with UINT64_MAX on s390x 64bit

### ISSUE THREAD

**Issue by jwhere** [zherczeg/sljit#265]
```c
#include <stdio.h>
#include <stdint.h>

#include "sljit/sljitLir.h"

typedef void(SLJIT_FUNC* fn)(unsigned char*);

int main(int argc, char* argv[]) {

unsigned char result[4] = { 0, };
fn f;
struct sljit_compiler* c = sljit_create_compiler(NULL);
struct sljit_jump* cmp, *out;
sljit_emit_enter(c, 0, SLJIT_ARGS1V(P), 1, 1, 0);

// result[0] = r0(imm 1) <= UINT64_MAX
sljit_emit_op1(c, SLJIT_MOV, SLJIT_R0, 0, SLJIT_IMM, 1);
cmp = sljit_emit_cmp(c, SLJIT_LESS_EQUAL, SLJIT_R0, 0, SLJIT_IMM, (sljit_uw)UINT64_MAX);
out = sljit_emit_jump(c, SLJIT_JUMP);
sljit_set_label(cmp, sljit_emit_label(c));
sljit_emit_op1(c, SLJIT_MOV_U8, SLJIT_MEM1(SLJIT_S0), 0, SLJIT_IMM, 1);
sljit_set_label(out, sljit_emit_label(c));

// result[1] = r0(imm 1) < UINT64_MAX
sljit_emit_op1(c, SLJIT_MOV, SLJIT_R0, 0, SLJIT_IMM, 1);
cmp = sljit_emit_cmp(c, SLJIT_LESS, SLJIT_R0, 0, SLJIT_IMM, (sljit_uw)UINT64_MAX);
out = sljit_emit_jump(c, SLJIT_JUMP);
sljit_set_label(cmp, sljit_emit_label(c));
sljit_emit_op1(c, SLJIT_MOV_U8, SLJIT_MEM1(SLJIT_S0), 1, SLJIT_IMM, 1);
sljit_set_label(out, sljit_emit_label(c));

// result[2] = r0(imm 1) >= UINT64_MAX
sljit_emit_op1(c, SLJIT_MOV, SLJIT_R0, 0, SLJIT_IMM, 1);
cmp = sljit_emit_cmp(c, SLJIT_GREATER_EQUAL, SLJIT_R0, 0, SLJIT_IMM, (sljit_uw)UINT64_MAX);
out = sljit_emit_jump(c, SLJIT_JUMP);
sljit_set_label(cmp, sljit_emit_label(c));
sljit_emit_op1(c, SLJIT_MOV_U8, SLJIT_MEM1(SLJIT_S0), 2, SLJIT_IMM, 1);
sljit_set_label(out, sljit_emit_label(c));

// result[3] = r0(imm 1) > UINT64_MAX
sljit_emit_op1(c, SLJIT_MOV, SLJIT_R0, 0, SLJIT_IMM, 1);
cmp = sljit_emit_cmp(c, SLJIT_GREATER, SLJIT_R0, 0, SLJIT_IMM, (sljit_uw)UINT64_MAX);
out = sljit_emit_jump(c, SLJIT_JUMP);
sljit_set_label(cmp, sljit_emit_label(c));
sljit_emit_op1(c, SLJIT_MOV_U8, SLJIT_MEM1(SLJIT_S0), 3, SLJIT_IMM, 1);
sljit_set_label(out, sljit_emit_label(c));

sljit_emit_return_void(c);
f = (fn)sljit_generate_code(c, 0, NULL);
f(result);
printf("%d %d %d %d\n", result[0], result[1], result[2], result[3]);
sljit_free_code(f, NULL);
sljit_free_compiler(c);
return 0;

}
```

## Expected Behavior
```
1 1 0 0
```
## Current Behavior
```
0 0 1 1
```

This only happens with UINT64_MAX.

**Comment by carenas** [comment]
if you add:

```
#include <limits.h>
```
so UINT64_MAX is defined, does it fix it?

**Comment by zherczeg** [comment]
You are right, this looks wrong. Probably a signed compare is used instead of unsigned. I can check it next week. If it is important, you can help me debugging it.

**Comment by carenas** [comment]
the following "fixes" the problem:

```
$ git diff
diff --git a/sljit_src/sljitNativeS390X.c b/sljit_src/sljitNativeS390X.c
index 3095d40..fd46aa3 100644
--- a/sljit_src/sljitNativeS390X.c
+++ b/sljit_src/sljitNativeS390X.c
@@ -2489,8 +2489,6 @@ static sljit_s32 sljit_emit_sub(struct sljit_compiler *compiler, sljit_s32 op,
                                        ins = (op & SLJIT_32) ? 0xc20f00000000 /* clfi */ : 0xc20e00000000 /* clgfi */;
                                        return emit_ri(compiler, ins, src1, src1, src1w, src2w, RIL_A);
                                }
-                               if (is_s16(src2w))
-                                       return emit_rie_d(compiler, 0xec00000000db /* alghsik */, (sljit_s32)tmp0, src1, src1w, src2w);
                        }
                }
                else if (src2 & SLJIT_MEM) {
```

**Comment by zherczeg** [comment]
Fixed it in 2c105e2461b0d5b6c9c632753522457ca442f9dd 

Unfortunately there was no better option than deleting it because of `sljit_set_current_flags`. The normal compare instructions sets the result differently than addition/subtraction. Please reopen this issue if the fix is not correct.
